lib/data/submission.js

```diff
@@ -28,15 +27,19 @@ const { union, last, pluck } = require('ramda');
 // if you are reading this thinking to use it elsewhere, you'll almost certainly
 // need to work out a sensible way to flag the SchemaStack allowEmptyNavigation boolean
 // to false for whatever you are doing.
-const submissionXmlToFieldStream = (fields, xml) => {
+const submissionXmlToFieldStream = (fields, xml, includeStructuralAttrs = false) => {
   const outStream = new Readable({ objectMode: true, read: noop });
 
   const stack = new SchemaStack(fields, true);
   let textBuffer = ''; // agglomerates text nodes that come as multiple events.
   const parser = new hparser.Parser({
-    onopentag: (name) => {
+    onopentag: (name, attrs) => {
       const field = stack.push(name);
-      if (field != null) { textBuffer = ''; }
+      if (field != null) {
+        textBuffer = '';
+        if (includeStructuralAttrs && attrs !== {} && !isEmpty(field) && field.isStructural())
```

Comment:
Won't `attrs !== {}` always be true? Should this be something like `Object.keys(attrs).length !== 0`?